pl.c

```diff
@@ -396,6 +413,52 @@ static void pl_cancel_add_jobs(struct playlist *pl)
 	worker_remove_jobs_by_cb(pl_match_add_job, pl);
 }
 
+/*
+ * (Re)load all files in cmus_playlist_dir.
+ *
+ * Walk cmus_playlist_dir; for each playlist file therein, either re-load the
+ * corresponding playlist in `pl_head' (if there is one), or load it (if there
+ * is not).
+ *
+ *
+ */
+
+void pl_reload(void)
```

Comment:
If I would call pl_reload(); at start instead of pl_load(); should't the result be the same? Is pl_load still necesery?

Reply:
I don't see a function named `pl_load`?

Reply:
Ah-- perhaps you mean `pl_load_all`? The two are similar, but different; `pl_load_all` assumes it is being called at startup, and so just runs through the files in the playlist directory, unconditionally creating new playlist structures for each one (via `pl_load_one`).  `pl_reload` has to take into account the possibility that there is already a playlist struct in-memory corresponding to each file processed.

Perhaps they could be merged, with a parameter to distinguish the two cases?

Reply:
Based on your description, shouldn't `pl_reload` just make `pl_load_all` obsolete?